### PR TITLE
Changes for the axsf implementation

### PR DESCRIPTION
**Pull request by zerothi**
Could you have a look at the changes I did here?

### CODE REVIEW

File: sisl/io/xsf.py [hunk 1]
```diff
@@ -219,44 +216,46 @@ def _r_geometry_multiple(self, steps, reuse_geom0=False, load_data=False, squeez
...
-        """ Returns Geometry object from the XSF file
+    def read_geometry(self, ret_data=False):
+        """ Geometry contained in file, and optionally the associated data
 
         Parameters
         ----------
-        data : bool, optional
-           in case the XSF file has auxiliary data, return that as well.
+        ret_data : bool, optional
+           in case the the file has auxiliary data, return that as well.
```
Comment (jonaslb) [inline sisl/io/xsf.py:256]: x2

**Comment by jonaslb** [comment]
I see that you removed the support for `one_geometry` -- do you think it is a waste? The reason I added it is because inelastica uses the axsf format for phononmodes, and the geometry is the same repeated over a thousand times (literally), its just the data that is different on each. So reading all the geometries is really a waste.

**Comment by zerothi** [comment]
> I see that you removed the support for `one_geometry` -- do you think it is a waste? The reason I added it is because inelastica uses the axsf format for phononmodes, and the geometry is the same repeated over a thousand times (literally), its just the data that is different on each. So reading all the geometries is really a waste.

Yeah, I think it is not really needed. The method still reads all of them, and I think it is wiser to do this outside the method. It is very easy to reduce on return.

**Comment by jonaslb** [comment]
Well, the method didn't *parse* all of it and didn't save the repeated information so it must have saved some work in the function. But in the end I guess it doesn't matter, people shouldn't use axsf for data storage anyway, i was just unfortunate to receive some previous calculation output in the format, but in this situation its still just a matter of doing a one-time conversion to a saner format, so I guess it doesn't matter if it takes a second longer or however much.

Ill merge this then, you can fix the typo later if you want ;)

**Comment by zerothi** [comment]
> Well, the method didn't _parse_ all of it and didn't save the repeated information so it must have saved some work in the function. But in the end I guess it doesn't matter, people shouldn't use axsf for data storage anyway, i was just unfortunate to receive some previous calculation output in the format, but in this situation its still just a matter of doing a one-time conversion to a saner format, so I guess it doesn't matter if it takes a second longer or however much.
> 
> Ill merge this then, you can fix the typo later if you want ;)

Thanks! Please fix the typo ;) While you're at it ;)

**Pull request merged**